### DISCUSSION TITLE
VolumeMeter class for SPL measurement (A & C Weighted)?

### DISCUSSION THREAD

**Discussion by SoundNerdsUnite** [pschatzmann/arduino-audio-tools#1771] [Q&A] [answered]
Hi!

I've been reading though the wonderful docs and came across the "VolumeMeter" function.
Is there a solution to measure and display the SPL result using this library? 

I came across[ Ivan's solution ](https://github.com/ikostoski/esp32-i2s-slm/blob/master/README.md)which seems promising.

Thanks for any help!

**Comment by pschatzmann** [comment, marked as answer]
The current implementation of the VolumeMeter is just using the max amplitude.
If you need something more sophisticated: Feel free to contribute.

**Reply by SoundNerdsUnite** [reply]
Great! Thank you, Phil!
